sqldelight-runtime/src/commonMain/kotlin/com/squareup/sqldelight/Transacter.kt

```diff
@@ -117,30 +115,13 @@ abstract class Transacter(private val database: SqlDatabase) {
   }
 
   abstract class Transaction {
-    internal val postCommitHooks: SharedSet<ThreadLocalRef<() -> Unit>> = SharedSet()
-    internal val postRollbackHooks: SharedSet<ThreadLocalRef<() -> Unit>> = SharedSet()
-    internal val queriesToUpdate: SharedSet<Query<*>> = SharedSet()
-
-    private val atomicSuccessful = AtomicBoolean(false)
-    internal var successful: Boolean
-      get() = atomicSuccessful.value
-      set(value) {
-        atomicSuccessful.value = value
-      }
-
-    private val atomicChildrenSuccessful = AtomicBoolean(true)
-    internal var childrenSuccessful: Boolean
-      get() = atomicChildrenSuccessful.value
-      set(value) {
-        atomicChildrenSuccessful.value = value
-      }
+    internal val postCommitHooks = sharedSet<() -> (() -> Unit)>()
```

Comment:
Seems like two type aliases might go a long way here so that it's `sharedSet<Supplier<Action>>>` or something

Reply:
yup good idea